### PR TITLE
Cleanup

### PR DESCRIPTION
**Pull request by flying-sheep**
Cleaned up everything.

Since this includes a few logic changes, please look at it before we merge.

and don’t start new work and branches not based on this, as everything will conflict.

### CODE REVIEW

File: R/kernel.r [hunk 1]
```diff
@@ -169,94 +192,100 @@ complete = function(request) {
...
-                        file_extension=".r", version=rversion
-                     ),
-                     banner=version$version.string
-                    )
-                )
+    rversion <- paste0(version$major, '.', version$minor)
+    send_response('kernel_info_reply', request, 'shell', list(
+        protocol_version       = '5.0',
+        implementation         = 'IRkernel',
+        implementation_version = '0.2',
```
Comment (JanSchulz) [inline R/kernel.r]: should this be 0.4?

Comment (takluyver) [inline R/kernel.r]: We should probably define a global version variable, so that it's easier to remember to update it.

File: R/execution.r [hunk 2]
```diff
@@ -1,12 +1,14 @@
 #' @include options.R
 NULL
 
-displayenv = environment(publish_mimebundle)
+setClassUnion('recordedplotOrNULL', members = c('recordedplot', 'NULL'))
```
Comment (takluyver) [inline R/execution.r]: @flying-sheep Can I suggest that functional changes to the code go in a different PR from the reformatting and style fixes? It's easier to review actual changes if they're not scattered through a big diff of trivial changes.

**Comment by flying-sheep** [comment]
Of course. That one was a mistake to change yet

**Comment by flying-sheep** [comment]
so can we pull this in?

**Comment by flying-sheep** [comment]
hey guys!

@takluyver 

@JanSchulz 

it would be very nice to use this as a base going on :smile:

**Comment by flying-sheep** [comment]
well, then i’ll simply do it :smile:

File: R/kernel.r (hunk 1 above)
Comment (flying-sheep) [inline R/kernel.r]: using `packageVersion` now

File: R/execution.r (hunk 2 above)
Comment (flying-sheep) [inline R/execution.r]: Of course. That one was a mistake to change yet

**Comment by takluyver** [comment]
Sorry, I forgot about this. I think it's all fine, though I'd still have preferred the functional change I commented on before to be a separate PR.

**Comment by flying-sheep** [comment]
ok, as said: it was a mistake, but worked quite well for some weeks for me.
